osu.Game.Rulesets.Osu/Mods/OsuModMirror.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModMirror : ModMirror, IApplicableToHitObject
+    {
+        public void ApplyToHitObject(HitObject hitObject)
+        {
+            var osuObject = (OsuHitObject)hitObject;
+
+            osuObject.Position = new Vector2(OsuPlayfield.BASE_SIZE.X - osuObject.X, osuObject.Position.Y);
+
+            if (!(hitObject is Slider slider))
+                return;
+
+            slider.NestedHitObjects.OfType<SliderTick>().ForEach(h => h.Position = new Vector2(OsuPlayfield.BASE_SIZE.X - h.Position.X, h.Position.Y));
+            slider.NestedHitObjects.OfType<SliderRepeat>().ForEach(h => h.Position = new Vector2(OsuPlayfield.BASE_SIZE.X - h.Position.X, h.Position.Y));
+
+            var controlPoints = slider.Path.ControlPoints.Select(p => new PathControlPoint(p.Position.Value, p.Type.Value)).ToArray();
+            foreach (var point in controlPoints)
+                point.Position.Value = new Vector2(-point.Position.Value.X, point.Position.Value.Y);
+
+            slider.Path = new SliderPath(controlPoints, slider.Path.ExpectedDistance.Value);
```

Comment:
Poor wording on my part in the PR description, this isn't the exact same as `OsuModHardRock`. The `Vector2` instances are changed to effectively 'reflect' over the y axis (to create a proper mirror) instead of the x ([as `OsuModHardRock` does](https://i.ppy.sh/93692827303c3f3743f9c94076e769b21d095122/68747470733a2f2f6f73752e7070792e73682f77696b692f696d616765732f47616d655f6d6f6469666965722f486172645f526f636b2f696d672f48522d636f6d70617269736f6e2d6f73752e6a7067)). Should I still try to abstract this as you suggested or would that be out of the scope of this PR?

Reply:
Yes, I will still consider this code duplication given the similarity between it and hard rock. Also for the fact that this will have a setting deciding whether to mirror horizontally or vertically, meaning both methods (or one method with a `Direction` enum) will have to exist.

Reply:
This isn't really that bad imo. There comes a point where you're overengineering things in the effort to eliminate all code duplication.

Reply:
Yeah that's a valid point, but for this case, I think since a utility class already exists and can have something like this, I think it's best to do that so you can have all similar logic grouped in one method shared between classes.